src/states/adult.rs

```diff
@@ -64,17 +66,19 @@ pub struct Adult {
     peer_map: PeerMap,
     peer_mgr: PeerManager,
     poke_timer_token: u64,
+    add_timer_token: u64,
     routing_msg_filter: RoutingMessageFilter,
     timer: Timer,
 }
 
 impl Adult {
```

Comment:
Be interesting how e identify `Infant`, i.e. an `Adult::age == 4` We perhaps do not need another state in the state machine (duplicate) but perhaps an `is_infant()` here?

Reply:
It might work. I would leave this issue for the design considerations after this PR, though, if that's okay.